src/coreComponents/codingUtilities/Table.hpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file Table.hpp
+ */
+
+#ifndef GEOS_COMMON_TABLE_HPP
+#define GEOS_COMMON_TABLE_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class Table
+{
+public:
+
+  enum Alignment { right, left, middle };
+
+  enum MarginType {border, column};
+
+  enum MarginValue { tiny, small, medium, large};
+
+  /**
+   * @brief Struct for a column margin and border margin.
+   * @param tiny 0 margin for both;
+   * @param small 1 margin from |
+   * @param medium 2 margin from |
+   * @param large 3 margin from |
+   *
+   * @param marginValue current margin value
+   */
+  struct Margin
+  {
+    integer tiny;
+    integer small;
+    integer medium;
+    integer large;
+
+    integer marginValue;
+
+    void setWorkingValue( integer const & value )
+    {
+      marginValue = value;
+    }
+  };
+
+  /**
+   * @brief Struct for a ColumnParam.
+   * @param headerName A vector containing all string for a header name
+   * @param columnValues Alignment for a column. By default aligned to the right side
+   * @param enabled A boolean to display a colummn
+   */
+  struct ColumnParam
+  {
+    std::vector< string > headerName;
+    Alignment alignment = Alignment::right;
+    bool enabled = true;
+  };
+
+  /**
+   * @brief Construct a new Table object, all values in the table are centered by default
+   *
+   * @param columnNames
+   */
+  Table( std::vector< string > const & columnNames );
+
+  /**
+   * @brief Construct a new Table object by specifying alignment of values in the table and optionally their display according to their log
+   * level
+   *
+   * @param columnParameter
+   */
+  Table( std::vector< ColumnParam > const & columnParameter );
+
+  /**
+   * @brief Add a row the the table. The number of arguments need to match with the number of header values
+   *
+   * @tparam N
+   * @tparam Args
+   * @param args
+   */
+  template< size_t N, typename ... Args >
+  void addRow( const Args &... args );
+
+  /**
+   * @brief Display the table
+   *
+   * @param os
+   */
+  void draw( std::ostream & os = std::cout );
+
+  /**
+   * @brief Set the name of the table
+   * @param [in] title The name of the table
+   */
+  void setTitle( string_view title );
+
+  /**
+   * @brief Set the Margin object
```

Comment:
Describe more precisely what it is doing : "Sets the minimal margin width between the row content an its borders."